railties/lib/rails/tasks/framework.rake

```diff
@@ -66,3 +68,24 @@ namespace :rails do
     end
   end
 end
+
+namespace :rails do
+  Rake.application.top_level_tasks.grep(/^rails:/).each do |task|
+    ActiveSupport::Deprecation.warn(<<-MSG.squish)
+      Running this task with the rails task namespace is deprecated in favor of app.
+      Please run e.g. bin/rails #{task.sub(/^rails:/, 'app:')} instead."
+    MSG
+  end
```

Comment:
I modified the codes for the :rails namespace in my local environment.

```ruby
namespace :rails do
  task :deprecated do |task|
    ActiveSupport::Deprecation.warn(<<-MSG.squish)
      Running #{task.name.sub('deprecated:', '')} with the rails: namespace is deprecated in favor of app.
      Run e.g. bin/rails app:update instead."
    MSG
  end

  task :update => ['deprecated', 'app:update']
  task :template => ['deprecated',  'app:template']

  namespace :templates do
    task :copy => ['deprecated', 'app:templates:copy']
  end

  namespace :update do
    task :configs => ['deprecated', 'app:update:configs']
    task :bin => ['deprecated', 'app:update:bin']
  end
end
```

and ran the command.

```
$ rails rails:update                                                                  1 ↵  
DEPRECATION WARNING: Running rails:deprecated with the rails: namespace is deprecated in favor of app. Run e.g. bin/rails app:update instead.". (called from block in execute at /Users/ryo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:238)
```

The message is incorrect because `rails:deprecated` is the original task used by the actual command.